pywbem_mock/_instancewriteprovider.py

```diff
@@ -277,114 +220,117 @@ def __init__(self, cimrepository=None):
+        - The namespace exists in the CIM repository.
+        - The creation class of the new instance exists in the namespace
+          in the CIM repository.
+        - All properties specified in the new instance are exposed by the
+          creation class in the CIM repository, and have the same CIM type and
+          array-ness.
 
-        1. Namespace is valid in CIM repository.
-        2. The ``NewInstance`` is a valid CIM Instance.
-        3. All key properties defined in the class  of ``NewInstance`` are
-           in ``NewInstance``. This method cannot add properties or
-           property values to ``NewInstance``.
-        4. There are no properties in ``NewInstance`` that are not in the
-           corresponding class.
-        5. NewInstance path must not exist in the CIM repository.
+        Validation that should be performed by the provider:
+        - Existence of all key properties in the new instance, if the provider
+          depends on that for determining the instance path.
+        - Non-existence of uninitializable properties in the new instance,
+          depending on the provider.
+        - Non-existence of the new CIM instance in the CIM repository.
 
         Parameters:
 
           namespace (:term:`string`):
-            The name of the CIM namespace in the CIM repository (case
-            insensitive). Must not be `None`. Leading or trailing slash
-            characters are ignored.
+            The name of the CIM namespace in which the CIM instance is to be
+            created, in any lexical case, and with leading and trailing slash
+            characters removed.
 
           NewInstance (:class:`~pywbem.CIMInstance`):
             A representation of the CIM instance to be created.
 
-            The ``classname`` attribute of this object specifies the creation
-            class for the new instance.
+            This object is a deep copy of the original client parameter, and may
+            be modified by the provider as needed, before storing it in the
+            CIM repository.
 
-            The namespace of the ``path`` attribute must be `None` or the
-            same as the namespace parameter.
+            The ``classname`` attribute of this object will specify the
+            creation class for the new instance, in any lexical case.
 
-            The `properties` attribute of this object specifies initial
-            property values for the new CIM instance.
+            The ``properties`` attribute of this object will specify the
+            initial property values for the new CIM instance, with property
+            names in any lexical case. Key properties may or may not be
+            included.
 
-            Instance-level qualifiers have been deprecated in CIM, so any
-            qualifier values specified using the `qualifiers` attribute
-            of this object are ignored.
+            The ``path`` attribute of this object will be `None`.
```

Comment:
I did not check yet.  Do we clear it in the validation?

Reply:
It is cleared in `WBEMConnection.CreateInstance()`